src/basic/pidref.c

```diff
@@ -260,6 +260,12 @@ int pidref_verify(PidRef *pidref) {
         return 1; /* We have a pidfd and it still points to the PID we have, hence all is *really* OK → return 1 */
 }
 
+int pidref_is_alive(PidRef *pidref) {
+        assert(pidref);
+
+        return pidref->fd >= 0 ? pidfd_is_alive(pidref->fd) : pid_is_alive(pidref->pid);
```

Comment:
oh, i just realized one more thing: pidref_is_alive() should return -ESRCH in case it is invoked with an argument initialized to PIDREF_NULL, i.e. where the pidfd is -EBADF and the pid is zero. But with this code we'd invoke pid_is_alive() with 0 as argument, which internally is mapped to our process ID, and we are of course always alive... 

hence this needs to handle the fd < 0 && pid == 0 case and return false.

Reply:
Hmm, I see the logic in `pid_is_alive()` now. I guess adding simple wrappers is not that simple....In any case, what is the rational for this kind of handling of pid `0` in pid_is_alive()?

Reply:
in the simple helpers that operate on processes by pid we systematically interpret pid=0 as "our own process", which is what classic UNIX kill() does too, and setpriority() as well and various other tools. It's simply how UNIX does this, and it's one of the things that make sense.